src/test/regress/expected/subselect_optimizer.out

```diff
@@ -1235,3 +1235,640 @@ drop table tl1;
+ <Column ColId=\"1\" Attno=\"-1\" ColName=\"ctid\"/>                                       +
+ <Column ColId=\"2\" Attno=\"-3\" ColName=\"xmin\"/>                                       +
+ <Column ColId=\"3\" Attno=\"-4\" ColName=\"cmin\"/>                                       +
+ <Column ColId=\"4\" Attno=\"-5\" ColName=\"xmax\"/>                                       +
+ <Column ColId=\"5\" Attno=\"-6\" ColName=\"cmax\"/>                                       +
+ <Column ColId=\"6\" Attno=\"-7\" ColName=\"tableoid\"/>                                   +
+ <Column ColId=\"7\" Attno=\"-8\" ColName=\"gp_segment_id\"/>                              +
+ </Columns>                                                                                +
+ </TableDescriptor>                                                                        +
+ </TableScan>                                                                              +
+ </GatherMotion>                                                                           +
+ </Sort>                                                                                   +
+ </Result>                                                                                 +
+ </Plan>"}
+(1 row)
+
+insert into t1 values (1);
+analyze t1;
+-- After adding data, the ORCA physical plan changes and DXL changes too.
+-- The outer <dxl:TestExpr> is empty. The second <dxl:TestExpr> (nested) does not contain a
+-- deep tree. The left node contains an attribute (Ident ColId="0") that is not present
+-- in the inner <dxl:SubPlan> node. Such attributes in <dxl:TestExpr> are treated as Vars.
+-- The previous case contained only params (attributes contained in the
+-- inner <dxl:SubPlan> node). This test aims to verify that <dxl:TestExpr> with params and
+-- Vars simultaneously is handled correctly during the DXL to Plan Statement stage.
+\! rm -rf $MASTER_DATA_DIRECTORY/minidumps
+set optimizer_trace_fallback=on;
+set optimizer_minidump=always;
+select * from t1 where
+  (i1 in (select i2 from t2)) in (select i3 = 1 from t3) order by i1;
+ i1 
+----
+  0
+  1
+(2 rows)
+
+reset optimizer_minidump;
+reset optimizer_trace_fallback;
+-- Output DXL plan without unimportant properties.
+-- start_ignore
+\! sed 's/></>\n</g' $MASTER_DATA_DIRECTORY/minidumps/*.mdp > $MASTER_DATA_DIRECTORY/minidumps/dump.xml
+\! sed -i '/Cost\|Properties\|ValuesList/d' $MASTER_DATA_DIRECTORY/minidumps/dump.xml
+\! sed -i 's/TypeMdid=".*"//' $MASTER_DATA_DIRECTORY/minidumps/dump.xml
+\! sed -i 's/AggMdid=".*"//' $MASTER_DATA_DIRECTORY/minidumps/dump.xml
+\! sed -i 's/SortOperatorMdid=".*"//' $MASTER_DATA_DIRECTORY/minidumps/dump.xml
+\! sed -i 's/OperatorMdid=".*"//' $MASTER_DATA_DIRECTORY/minidumps/dump.xml
+\! sed -i 's/Mdid=".*"//' $MASTER_DATA_DIRECTORY/minidumps/dump.xml
+\! sed -i 's/dxl://' $MASTER_DATA_DIRECTORY/minidumps/dump.xml
+-- end_ignore
+select xpath ('.//Plan', xmlparse(document pg_read_file('minidumps/dump.xml')));
```

Comment:
Can we use PL/pgSQL to indent output of xpath? It is difficult to read and understand the current version of the xml.